csharp/ql/src/semmle/code/csharp/ir/implementation/raw/internal/desugar/Foreach.qll

```diff
@@ -206,17 +208,14 @@ private class TranslatedForeachGetEnumerator extends TranslatedCompilerGenerated
 
   override Callable getInstructionFunction(InstructionTag tag) {
     tag = CallTargetTag() and
-    exists(Callable internal |
-      internal.getName() = "GetEnumerator" and
-      // TODO: For now ignore the possibility that the
-      //       foreach variable can have a generic type.
-      //       The type of the callable will need to be fabricated,
-      //       since we might not find the correct callable in the DB.
-      //       Probably will have change the way the immediate
-      //       operand of `FunctionAddress` is calculated.
-      internal.getReturnType().getName() = "IEnumerator" and
-      result = internal
-    )
+    result.getName() = "GetEnumerator" and
```

Comment:
I think this predicate will return *any* `GetEnumerator` method, not the specific one that is actually called. Can we look at the type of the expression in order to ensure that there is exactly one call target (even if it's just `IEnumerable<T>.GetEnumerator()`).